server/expression/linting/missing_variables.go

```diff
@@ -0,0 +1,118 @@
+package linting
+
+import (
+	"reflect"
+	"time"
+
+	"github.com/kubeshop/tracetest/server/expression"
+)
+
+func DetectMissingVariables(target interface{}, availableVariables []string) []string {
+	missingVariables := []string{}
+
+	traverse(target, func(f reflect.StructField, value string) {
+		tokens := getTokens(f, value)
+		variables := make([]string, 0)
+		for _, token := range tokens {
+			if token.Type == expression.EnvironmentType {
+				variables = append(variables, token.Identifier)
+			}
+		}
+
+		missingVariables = append(missingVariables, getSetDifference(variables, availableVariables)...)
+	})
+
+	uniqueMissingVariables := make([]string, 0)
+	existingMissingVariables := make(map[string]bool)
+
+	for _, variable := range missingVariables {
+		if _, exists := existingMissingVariables[variable]; !exists {
+			uniqueMissingVariables = append(uniqueMissingVariables, variable)
+			existingMissingVariables[variable] = true
+		}
+	}
+
+	return uniqueMissingVariables
+}
+
+func getTokens(fieldInfo reflect.StructField, value string) []expression.ReflectionToken {
+	if fieldInfo.Tag.Get("expr_enabled") == "true" {
+		exprTokens, err := expression.GetTokensFromExpression(value)
```

Comment:
So we go through all of the fields from a struct and depending on the type we decide if we recursively traverse over it or we just land into callback which triggers the getTokens function at some point.

Is this a standard way of traversing objects in golang?